modules/search-pipeline-common/src/main/java/org/opensearch/search/pipeline/common/OversampleRequestProcessor.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.search.pipeline.common;
+
+import org.opensearch.action.search.SearchRequest;
+import org.opensearch.ingest.ConfigurationUtils;
+import org.opensearch.search.pipeline.AbstractProcessor;
+import org.opensearch.search.pipeline.Processor;
+import org.opensearch.search.pipeline.SearchRequestProcessor;
+import org.opensearch.search.pipeline.StatefulSearchRequestProcessor;
+import org.opensearch.search.pipeline.common.helpers.ContextUtils;
+
+import java.util.Map;
+
+import static org.opensearch.search.pipeline.common.helpers.ContextUtils.applyContextPrefix;
+
+/**
+ * Multiplies the "size" parameter on the {@link SearchRequest} by the given scaling factor, storing the original value
+ * in the request context as "original_size".
+ */
+public class OversampleRequestProcessor extends AbstractProcessor implements StatefulSearchRequestProcessor {
+
+    /**
+     * Key to reference this processor type from a search pipeline.
+     */
+    public static final String TYPE = "oversample";
+    static final String SAMPLE_FACTOR = "sample_factor";
+    static final String ORIGINAL_SIZE = "original_size";
+    private final double sampleFactor;
+    private final String contextPrefix;
+
+    private OversampleRequestProcessor(String tag, String description, boolean ignoreFailure, double sampleFactor, String contextPrefix) {
+        super(tag, description, ignoreFailure);
+        this.sampleFactor = sampleFactor;
+        this.contextPrefix = contextPrefix;
+    }
+
+    @Override
+    public SearchRequest processRequest(SearchRequest request, Map<String, Object> requestContext) {
+        if (request.source() != null) {
+            int originalSize = request.source().size();
```

Comment:
Question: as size is optional parameter. when control this code path does the size already set to default value of 10?

Reply:
Good catch!

The default of 10 is set by `DefaultSearchContext`: https://github.com/opensearch-project/OpenSearch/blob/e5c4f9d2016bfa50c464945dc9533aa4aee49922/server/src/main/java/org/opensearch/search/DefaultSearchContext.java#L258

That executes after this. By default, this will get `originalSize = -1`.

I guess I can check for -1 and initialize it to 10 instead, though that's duplicating the logic from `DefaultSearchContext`. I can also move it out into a constant -- I'm a little surprised it's not already a constant.

Reply:
Oh -- actually it is a constant, just not used by DefaultSearchContext:

https://github.com/opensearch-project/OpenSearch/blob/e98ded62d4b672a03acb4beb02bd047d7d4e1c69/server/src/main/java/org/opensearch/search/SearchService.java#L272

I think I'll change DefaultSearchContext to use that too.